module/data/activity/base-activity.mjs

```diff
@@ -146,6 +146,244 @@ export default class BaseActivityData extends foundry.abstract.DataModel {
+    return {
+      type: source.system.activation?.type === "none" ? "" : (source.system.activation?.type ?? ""),
+      value: source.system.activation?.cost ?? null,
+      condition: source.system.activation?.condition ?? ""
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetch activation data from the item source and transform it into an activity's consumption object.
+   * @param {object} source  Item's candidate source data to transform.
+   * @returns {object}       Creation data for new activity.
+   */
+  static transformConsumptionData(source) {
+    const targets = [];
+
+    const type = {
+      attribute: "attribute",
+      hitDice: "hitDice",
+      material: "material",
+      charges: "itemUses"
+    }[source.system.consume?.type];
+
+    if ( type ) targets.push({
+      type,
+      target: source.system.consume?.target ?? "",
+      value: source.system.consume?.amount ?? "1",
+      scaling: {
+        mode: source.system.consume?.scale ? "amount" : "",
+        formula: ""
+      }
+    });
+
+    // If no target type set but this item has max uses, set consumption type to itemUses with blank target
+    else if ( source.system.uses?.max ) targets.push({
+      type: "itemUses",
+      target: "",
+      value: "1",
+      scaling: {
+        mode: source.system.consume?.scale ? "amount" : "",
+        formula: ""
+      }
+    });
+
+    return {
+      targets,
+      scaling: {
+        allowed: source.system.consume?.scale ?? false,
+        max: ""
+      }
+    };
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Fetch action data from the item source and transform it into an activity's damage parts array.
+   * @param {object} source  Item's candidate source data to transform.
+   * @param {string[]} part  The damage part to transform.
+   * @returns {object}       Creation data for new activity.
+   */
+  static transformDamagePartData(source, [formula, type]) {
+    const data = {
+      number: null,
+      denomination: null,
+      bonus: "",
+      types: type ? [type] : [],
+      custom: {
+        enabled: false,
+        formula: ""
+      },
+      scaling: {
+        mode: source?.system.scaling?.mode !== "none" ? "whole" : "",
+        number: null,
+        formula: source?.system.scaling?.formula ?? ""
+      }
+    };
+
+    const parsed = (formula ?? "").match(/^\s*(\d+)d(\d+)(?:\s*([+|-])\s*(@?[\w\d.]+))?\s*$/i);
+    if ( parsed && CONFIG.DND5E.dieSteps.includes(Number(parsed[2])) ) {
+      data.number = Number(parsed[1]);
+      data.denomination = Number(parsed[2]);
+      if ( parsed[4] ) data.bonus = parsed[3] === "-" ? `-${parsed[4]}` : parsed[4];
+    } else {
+      data.custom.enabled = true;
+      data.custom.formula = formula;
+    }
+
+    // If scaling denomination matches the damage denomination, set scaling using number rather than formula
+    const scaling = data.scaling.formula.match(/^\s*(\d+)d(\d+)\s*$/i);
```

Comment:
Same as above, it feels like unnecessary work/risk when we can just set a formula in both instances and be done. If you are confident that we won't incorrectly migrate something and lose data though, then that's fine.